edb/lib/std/20-genericfuncs.edgeql

```diff
@@ -321,6 +333,7 @@ std::max(vals: SET OF duration) -> OPTIONAL duration
         'Return the greatest value of the input set.';
     SET volatility := 'Immutable';
     SET force_return_cast := true;
+    SET preserves_optionality := true;
```

Comment:
Please add a check to `edb/schema/function.py` to check if all variants of function specify `preserves_optionality` if one sets it.

Reply:
Good call.  I actually planned to add a whole bunch of validation, but somehow forgot to do it :man_facepalming:

Reply:
We would be able to support them not doing it, though, right? Disallowing it would be because we don't think we'd ever want to on purpose?

Reply:
Semantically, all overloads should behave the same, I can't think of a use case where a specific overload would need to be exempted.

Reply:
Fixed.